server/middleware/validationMiddleware.ts

```diff
@@ -32,6 +32,11 @@ function validationMiddleware(conditionService: ConditionService, type?: new ()
         { excludeExtraneousValues: false }
       )
 
+      const { appointmentTimeType } = req.body
+      if (!!appointmentTimeType && appointmentTimeType !== 'SPECIFIC_DATE_TIME') {
```

Comment:
this is to skip date & time validation when appointment time type is other than Specific date time.

Reply:
this is a bit of a code smell - having bespoke logic in otherwise generic code. 

Is it possible to add a custom validator or someother mechanism to add conditional validation?